api/api/test/test_authentication.py

```diff
@@ -72,44 +72,45 @@ def test_check_user(mock_raise_if_exc, mock_submit, mock_distribute_function, mo
     mock_raise_if_exc.assert_called_once()
 
 
-@patch('api.authentication.token_urlsafe', return_value='test_token')
+@patch('api.authentication.change_keyPair', return_value=["b'-----BEGIN PRIVATE KEY-----\n'", "b'-----BEGIN PUBLIC KEY-----\n'"])
 @patch('os.chmod')
-@patch('api.authentication.chown')
+@patch('os.chown')
 @patch('builtins.open')
-def test_generate_secret(mock_open, mock_chown, mock_chmod, mock_token):
-    """Check if result's length and type are as expected and mocked function are called with correct params."""
-    result = authentication.generate_secret()
-    assert isinstance(result, str)
-    assert result == 'test_token'
-
-    calls = [call(authentication._secret_file_path, mode='x')]
-    mock_open.has_calls(calls)
-    mock_chown.assert_called_once_with(authentication._secret_file_path, authentication.wazuh_uid(),
-                                       authentication.wazuh_gid())
-    mock_chmod.assert_called_once_with(authentication._secret_file_path, 0o640)
+def test_generate_keyPair(mock_open, mock_chown, mock_chmod, mock_change_keyPair):
+    """Verify correct params when calling open method inside generate_keyPair"""
+    result = authentication.generate_keyPair()
+    assert isinstance(result[0], str)
+    assert isinstance(result[1], str)
+    assert result == ["b'-----BEGIN PRIVATE KEY-----\n'", "b'-----BEGIN PUBLIC KEY-----\n'"]
```

Comment:
The result will be `["---....", "-----....."]` (without 'b') as we are reading without the `b` flag. This will change depending on the decision taken in one of the last comments where this was commented.

The tests that follow will need to be changed too.